### PR TITLE
Receives values as String list only.

### PR DESCRIPTION
**Pull request by jung-hunsoo**
I've recognized that your original codes are intended to receive values as either atoms or strings.

But unfortunately the string values produce failure on Atom.to_string/1, so there are two remedies;
1) Cast atom to string selectively in the module using Enum.map/2.
2) Receive values as a string list only. (Developers will be forced not to use atom type)

At first I chose 1), but practically there's no advantages to use non-GCed atom type more than necessary. So I condensed this exnumerator to put more simple to use on.

### CODE REVIEW

File: lib/exnumerable.ex [hunk 1]
```diff
@@ -2,19 +2,17 @@ defmodule Exnumerator do
   defmacro __using__(opts) do
     quote location: :keep, bind_quoted: [values: opts[:values]] do
       @behaviour Ecto.Type
-
+
```
Comment (KamilLelonek) [inline lib/exnumerable.ex]: Do we need these empty spaces here?

File: lib/exnumerable.ex [hunk 2]
```diff
@@ -2,19 +2,17 @@ defmodule Exnumerator do
...
-        def load(unquote(string)), do: {:ok, unquote(atom)}
-        def dump(unquote(string)), do: {:ok, unquote(string)}
-        def dump(unquote(atom)),   do: {:ok, unquote(string)}
+      for value <- values do
+        def cast(unquote(value)), do: {:ok, unquote(value)}
+        def load(unquote(value)), do: {:ok, unquote(value)}
+        def dump(unquote(value)), do: {:ok, unquote(value)}
       end
-
+
```
Comment (KamilLelonek) [inline lib/exnumerable.ex]: Do we need these empty spaces here?

File: test/exnumerable_test.exs [hunk 3]
```diff
@@ -14,20 +14,18 @@ defmodule ExnumeratorTest do
...
-    assert Message.cast(:read)       == {:ok, :read}
-    assert Message.load("received")  == {:ok, :received}
+  test "should argument given types" do
+    assert Message.cast("sent")      == {:ok, "sent"}
+    assert Message.load("received")  == {:ok, "received"}
     assert Message.dump("delivered") == {:ok, "delivered"}
-    assert Message.dump(:sent)       == {:ok, "sent"}
+    assert Message.cast(:sent)       == :error
+    assert Message.load(:received)   == :error
+    assert Message.dump(:delivered)  == :error
```
Comment (KamilLelonek) [inline test/exnumerable_test.exs]: These errors should not be handled in this test, shouldn't they? We are checking for positive results here. The other test checks for errors.

File: test/exnumerable_test.exs [hunk 4]
```diff
@@ -14,20 +14,18 @@ defmodule ExnumeratorTest do
...
   end
 
   test "should not cast unknown argument" do
     assert Message.cast("invalid") == :error
-    assert Message.cast(:invalid)  == :error
     assert Message.load("invalid") == :error
-    assert Message.load(:invalid)  == :error
     assert Message.dump("invalid") == :error
-    assert Message.dump(:invalid)  == :error
```
Comment (KamilLelonek) [inline test/exnumerable_test.exs]: Why are these removed? Do we still allow to handle atoms?

**Review by KamilLelonek — CHANGES_REQUESTED** [review]

File: lib/exnumerable.ex (hunk 1 above)
Comment (jung-hunsoo) [reply lib/exnumerable.ex]: Not really.

File: lib/exnumerable.ex (hunk 2 above)
Comment (jung-hunsoo) [reply lib/exnumerable.ex]: of course not.

File: test/exnumerable_test.exs (hunk 3 above)
Comment (jung-hunsoo) [reply test/exnumerable_test.exs]: This is actually the most important part, since this branch will break backward compatibility. Users no longer pass atom lists.
But you're right. This test specs won't be asserted. Test specs in their individual apps should assert the changes.

Comment (KamilLelonek) [reply test/exnumerable_test.exs]: I don't want to remove this test but to separate successful checks from the errored ones in a separate example.

**Review by KamilLelonek — APPROVED** [review]

**Comment by jung-hunsoo** [comment]
It would be better to consider up-numbering version as well, since this PR will break backward compatibility.

**Comment by KamilLelonek** [comment]
Can you bum the version number then?

**Comment by jung-hunsoo** [comment]
Current version is "1.2.2". Next can be "1.2.3" or "1.3.0".

"1.2.3" is reasonable when this PR is considered as a mere functional change.
"1.3.0" is reasonable when this PR breaks backward-compatibility.

Which one do you think is more fitted?

**Comment by KamilLelonek** [comment]
It does break, doesn't it?

**Comment by jung-hunsoo** [comment]
You're right. It breaks inevitably.

**Comment by KamilLelonek** [comment]
Can you update version number then?
